performance_regression_gradual_grow_throughput.py

```diff
@@ -182,45 +204,55 @@ def run_gradual_increase_load(self, stress_cmd_templ,
-        popuplation_distribution = CSPopulateDistribution(self.params.get("cs_populating_distribution"))
-        if popuplation_distribution == CSPopulateDistribution.GAUSS:
-            return f"GAUSSIAN(1..{self.ROW_COUNT},{self.GAUSS_CENTER},{self.GAUSS_SIGMA})"
-        elif popuplation_distribution == CSPopulateDistribution.UNIFORM:
-            return f"UNIFORM(1..{self.ROW_COUNT})"
-        else:
-            self.log.error("Unsupported cs population distribution")
-            return ""
+        self.log.debug("es_index: %s", self._test_index)
+        self.log.debug("total_summary: %s", total_summary)
+        self.log.debug("setup_details: %s", setup_details)
+        is_gce = bool(self.params.get('cluster_backend') == 'gce')
+        perf_analyzer.check_regression(test_id=self.test_id, test_name=test_name, subtest_baseline=sub_type,
+                                       test_results=total_summary, test_details=setup_details, is_gce=is_gce)
 
     @staticmethod
-    def _calculate_average_latency(results):
+    def _calculate_average_max_latency(results):
         status = defaultdict(float).fromkeys(results[0].keys(), 0.0)
+        max_latency = defaultdict(list)
+
         for result in results:
             for key in status:
                 try:
                     status[key] += float(result.get(key, 0.0))
+                    if key in ["latency 95th percentile", "latency 99th percentile"]:
+                        max_latency[f"{key} max"].append(float(result.get(key, 0.0)))
                 except ValueError:
                     continue
 
         for key in status:
             status[key] = round(status[key] / len(results), 2)
 
+        for key, latency in max_latency.items():
+            status[key] = max(latency)
+
         return status
 
-    def warmup_cache(self, compaction_strategy):
-        cmd = f"cassandra-stress read no-warmup cl=QUORUM duration=180m -pop 'dist={self.get_cs_distribution()}' -mode native cql3 -rate 'threads=500 throttle=35000/s'"  # pylint: disable=line-too-long
-        stress_queue = self.run_stress_cassandra_thread(
-            stress_cmd=cmd,
-            stress_num=1,
-            compaction_strategy=compaction_strategy,
-            stats_aggregate_cmds=False,
-            round_robin=True
-        )
-        self.get_stress_results(stress_queue, store_results=False)
+    def warmup_cache(self, stress_cmd_templ):
+        stress_queue = []
+        for stress_cmd in stress_cmd_templ:
+            params = {"round_robin": True, "stats_aggregate_cmds": False}
+            stress_cmd_to_run = stress_cmd.replace("$threads", "1000")
+            # Run c-s command without throttle
+            stress_cmd_to_run = stress_cmd_to_run.replace("$throttle", "")
+            # TODO: get "n" from c-s command or do it configurable
+            stress_cmd_to_run = stress_cmd_to_run.replace(
+                "duration=20m", "n=216666669" if "gauss" in stress_cmd_to_run else "n=20000000")
```

Comment:
I'm not sure I understand the else part: "else "n=20000000"".
What is this 20M?

Reply:
It is for read test. We run smaller data set for read

Reply:
but it's not 20M, it's 5M for each command, isn't it?